### PR TITLE
feat(file): add xcp_readlink function

### PR DESCRIPTION
**Pull request by Wescoeur**

### CODE REVIEW

**Comment by benjamreis** [comment]
Add file.h include in global header.

**Comment by Wescoeur** [comment]
Oh! Good point!

File: src/file.c [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  while ((size_t)(ret = readlink(pathname, buf, bufSize)) == bufSize) {
+    char *p = realloc(buf, bufSize << 1);
+    if (!p) {
+      free(buf);
+      return NULL;
+    }
+    buf = p;
+  }
+
+  if (ret == -1) {
```
Comment (benjamreis) [inline src/file.c:39]: Log error and `errno`.

**Comment by benjamreis** [comment]
ok

**Pull request merged**